Content.Shared/_Starlight/Clothing/EntitySystems/HardsuitChemicalImmunitySystem.cs

```diff
@@ -0,0 +1,98 @@
+using Content.Shared._Starlight.Chemistry.Events;
+using Content.Shared._Starlight.Clothing.Components;
+using Content.Shared.Damage.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Popups;
+using Robust.Shared.Network;
+using InventoryComponent = Content.Shared.Inventory.InventoryComponent;
+
+namespace Content.Shared._Starlight.Clothing.EntitySystems;
+
+/// <summary>
+/// System that handles hardsuit chemical immunity, preventing injection-based attacks
+/// when wearing hardsuits with the immunity component.
+/// </summary>
+public sealed class HardsuitChemicalImmunitySystem : EntitySystem
+{
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly INetManager _net = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        
+        // Subscribe to injection attempt events to check for hardsuit immunity
+        // For melee injections (wonderprod)
+        SubscribeLocalEvent<InventoryComponent, InjectOnHitAttemptEvent>(OnInventoryMeleeInjectAttempt);
+        SubscribeLocalEvent<HardsuitChemicalImmunityComponent, InjectOnHitAttemptEvent>(OnHardsuitMeleeInjectAttempt);
+        
+        // For projectile injections (tranquilizer shells)
+        SubscribeLocalEvent<InventoryComponent, SolutionInjectAttemptEvent>(OnInventoryProjectileInjectAttempt);
+        SubscribeLocalEvent<HardsuitChemicalImmunityComponent, SolutionInjectAttemptEvent>(OnHardsuitProjectileInjectAttempt);
+        
+    }
+
+    // Melee injection handlers (wonderprod)
+    private void OnInventoryMeleeInjectAttempt(EntityUid uid, InventoryComponent component, ref InjectOnHitAttemptEvent args)
+    {
+        // Check the outerClothing slot for hardsuit immunity
+        if (_inventory.TryGetSlotEntity(uid, "outerClothing", out var outerClothing, component))
+        {
+            RaiseLocalEvent(outerClothing.Value, ref args, true);
+        }
+    }
+
+    private void OnHardsuitMeleeInjectAttempt(Entity<HardsuitChemicalImmunityComponent> ent, ref InjectOnHitAttemptEvent args)
```

Comment:
Did you know that doing a melee attack with cuffs is raising InjectOnHitAttemptEvent? is this making people wearing hardsuits immune to being cuffed with melee attacks?